plugins/jenkins/app/models/samson/jenkins.rb

```diff
@@ -30,17 +52,144 @@ def initialize(job_name, deploy)
       @deploy = deploy
     end
 
+    def jenkins_job_config
+      conf = Rails.cache.fetch(job_name + "_conf", expires_in: 1.day, race_condition_ttl: 5.minute) do
+        client.job.get_config(job_name)
+      end
+      Nokogiri::XML(conf)
+    end
+
+    # we need to add new build parameters without manually changing each job's configuration in
+    # manually. To do this, we check if this job is mentioned in
+    # description of jenkins job. if not, we add it. We use JENKINS_DESC_HEADER/FOOTER markers
+    # to keep track of description added through samson.
+    def send_job_config
+      conf = jenkins_job_config
+      post_config = false
+      expected_build_parameters = JENKINS_BUILD_PARAMETERS.keys
+      present_build_params = build_params(conf).map(&:to_sym)
+      missing_build_params = expected_build_parameters.to_set - present_build_params.to_set
+      if missing_build_params.any?
+        add_build_parameters(conf, missing_build_params)
+        post_config = true
+      end
+      unless description_exists?(conf)
+        add_job_description(conf)
+        post_config = true
+      end
+      if post_config
+        post_job_config(conf)
+      end
+    end
+
+    def build_params(conf)
+      params = conf.xpath("//hudson.model.StringParameterDefinition")
+      params_array = []
+      params.each do |param|
+        param.children.each do |value|
+          if value.name == "name"
+            if val = value.content.split(JENKINS_BUILD_PARAMETRS_PREFIX, 2)[1]
+              params_array << val
+            end
+          end
+        end
+      end
+      params_array
+    end
+
+    def description(conf)
+      conf.xpath("//description").first
+    end
+
+    def jenkins_desc_job_name
+      "* #{deploy.project.name} - #{deploy.stage.name}"
+    end
+
+    def extract_description(conf)
+      job_names = []
```

Comment:
both work, I was thinking it would look more symmetrical and code readers would ahve to keep less variables in mind if the code was:

```Ruby
a = 1

if foo
  a = 2
  b = 3
else
  a = 3
  b = 4
end
[a, b]
```